packages/lib/bookingSuccessRedirect.ts

```diff
@@ -8,23 +8,34 @@ import { useCompatSearchParams } from "@calcom/lib/hooks/useCompatSearchParams";
 import { navigateInTopWindow } from "@calcom/lib/navigateInTopWindow";
 
 function getNewSearchParams(args: {
-  query: Record<string, string | null | undefined | boolean>;
+  query: Record<
+    string,
+    string | null | undefined | boolean | { name?: string } | Array<{ name: string; email?: string }>
+  >;
   searchParams?: URLSearchParams;
 }) {
   const { query, searchParams } = args;
   const newSearchParams = new URLSearchParams(searchParams);
+
   Object.entries(query).forEach(([key, value]) => {
-    if (value === null || value === undefined) {
-      return;
+    if (value === null || value === undefined) return;
+
+    if (Array.isArray(value) && key === "attendees" && value.length > 0) {
```

Comment:
I think we shouldn't make `getNewSearchParams` aware of application logic.

Reply:
Hmm. Yeah that's a good point

Reply:
I think we could  update `getBookingRedirectExtraParams` to return what we need including the transformation(user -> hostName, attendees -> attendeeName)

Reply:
We could also send guests as well

Reply:
Yeah 👍